### ISSUE TITLE
Allow passing null values in Custom JSON Scalars

### ISSUE THREAD

**Issue by DavidKmn** [apollographql/apollo-android#2920]
We have a custom JSON Scalar type that we need represented, I have looked through #1821 and tried the suggested however we are getting an issue with the generated code. See the below screenshot:
![Screenshot 2021-02-09 at 15 41 12](https://user-images.githubusercontent.com/35953014/107388439-c3d14f80-6aed-11eb-8e25-31892d8d6d9c.png)

I am also attaching other relevant pieces to help with this investigation. 
![Screenshot 2021-02-09 at 15 41 23](https://user-images.githubusercontent.com/35953014/107388556-df3c5a80-6aed-11eb-9610-914e62946883.png)
![Screenshot 2021-02-09 at 15 41 52](https://user-images.githubusercontent.com/35953014/107388580-e499a500-6aed-11eb-8ae6-aae32b81ba4f.png)
![Screenshot 2021-02-09 at 15 41 32](https://user-images.githubusercontent.com/35953014/107388604-eb281c80-6aed-11eb-83ed-a455072f89d9.png)

I know you @sav007 tried to help with this previously, could you please let me know what you think?

**Comment by martinbonnin** [comment]
Hi @DavidKmn 👋 , Apollo Android unfortunately doesn't support generic types for custom scalars. As a workaround, you can wrap your Map in another class:

```
class MyJson(val map:Map<String, Any?>)
```
```
customTypeMapping = [ "JSONScalar": "com.example.MyJson"]
```

**Comment by DavidKmn** [comment]
Thanks for the prompt response @martinbonnin, what would the Type Adapter look like then? 
![Screenshot 2021-02-09 at 16 18 07](https://user-images.githubusercontent.com/35953014/107393179-668bcd00-6af2-11eb-9bf8-e67495956c4e.png)

**Comment by martinbonnin** [comment]
Ooooohoo, this looks like an oversight actually 👀 . I don't think assuming non-null values in `GraphQLJsonObject` is correct. I'll look into it. 

In the meantime, you can make everything non-nullable:

```
class MyJson(val map:Map<String, Any>)
```

I haven't checked but I think this would also work if for some reason some data is null (mainly because there are unsafe casts later on in the runtime. This is not great...). If you data is all non-nullable, it should be fine in all cases.

**Comment by DavidKmn** [comment]
Yeah, though so. Thank you. Will be looking out for an update :) In the meantime will give your suggestion a try

**Comment by martinbonnin** [comment]
Thanks, let me know how that goes!

**Comment by martinbonnin** [comment]
Hi @DavidKmn sorry for the silence. How did that go? Were you able to use your custom scalar? 

As a heads up, `3.0.0-alpha01` is out, that reworks the `Adapter` API to allow nullable values.

**Comment by martinbonnin** [comment]
Closing as fixed in `3.0.0-alpha01`
